### PR TITLE
Move public exports from `exports.py` to `backend/__init__py`.

### PR DESCRIPTION
**Pull request by hertschuh**
The main motivation is to use the exported `keras.Variable` class throughout code.
The exported `keras.Variable` class is a subclass of the backend specific implementation. However, that public class was not used internally to create variables, for instance via `add_weight()`.
There was no convenient way to detect that a tensor really is a variable. This code would print `False`:
```python
v = self.add_weight(x)
print(isinstance(v, keras.Variable)
```

Additionally, made API exposed in `keras/src/backend/<backend>/__init__.py` consistent between each other and consistent with `keras.src.backend`, which is important when using the `DynamicBackend` feature:
- added `device_scope()` to Numpy backend (a no-op)
- added import for `name_scope()` in all backends
- fixed bug where `SeedGenerator` was not using the dynamic backend specific `name_scope`

### CODE REVIEW

File: keras/src/random/seed_generator.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def seed_initializer(*args, **kwargs):
             return self.backend.convert_to_tensor([seed, 0], dtype=dtype)
 
         with backend.name_scope(self.name, caller=self):
-            self.state = self.backend.Variable(
+            self.state = self.backend.BackendVariable(
```
Comment (fchollet) [inline keras/src/random/seed_generator.py:75]: Can you explain why this is necessary?

File: keras/src/backend/__init__.py [hunk 2]
```diff
@@ -47,3 +48,8 @@
     distribution_lib = None
 else:
     raise ValueError(f"Unable to import backend : {backend()}")
+
+
+@keras_export("keras.Variable")
```
Comment (fchollet) [inline keras/src/backend/__init__.py:53]: Please delete `exports.py` and move all the exports to the end of the file here, that's cleaner. No need to make Variable an exception

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/src/backend/torch/core.py [hunk 3]
```diff
@@ -187,11 +187,7 @@ def __eq__(self, other):
 def convert_to_tensor(x, dtype=None, sparse=None):
     if sparse:
         raise ValueError("`sparse=True` is not supported with torch backend")
-    if type(x) is Variable:
-        # We cannot use `isinstance(x, Variable)` due to the failure of
```
Comment (hertschuh) [inline keras/src/backend/torch/core.py:191]: It appears to be fixed based on the fact that the unit tests and CI tests pass. See discussion here: https://github.com/keras-team/keras/pull/19955#discussion_r1790558245

File: keras/src/backend/__init__.py (hunk 2 above)
Comment (hertschuh) [reply keras/src/backend/__init__.py:55]: Done.

File: keras/src/random/seed_generator.py (hunk 1 above)
Comment (hertschuh) [reply keras/src/random/seed_generator.py]: Reverted by using a different approach.

**Comment by hertschuh** [comment]
Updated for all of `exports.py`

**Review by fchollet — APPROVED** [review]

**Pull request merged**